src/tasks/gitea/gitea-drone-oauth.test.ts

```diff
@@ -0,0 +1,146 @@
+  })
+  it('should fail validation when the secret contains invalid data', () => {
+    const stubRemoteSecret: DroneSecret = {
+      clientId: Buffer.from('abc').toString('base64'),
+      clientSecret: '',
+    }
+    const stubOauthApps: OAuth2Application[] = [
+      {
+        name: 'otomi-drone',
+        clientId: 'abc',
+      },
+    ]
+    expect(isSecretValid(stubRemoteSecret, stubOauthApps)).to.be.false
+    expect(consoleLog.calledWith('Remote secret values were empty')).to.be.true
+  })
+  it('should fail to validate on empty OAuthApp list', () => {
+    const stubRemoteSecret: DroneSecret = {
+      clientId: Buffer.from('abc').toString('base64'),
+      clientSecret: Buffer.from('def').toString('base64'),
+    }
+    const stubOauthApps: OAuth2Application[] = []
+    expect(isSecretValid(stubRemoteSecret, stubOauthApps)).to.be.false
+    expect(consoleLog.calledWith("Gitea doesn't have any oauth application defined")).to.be.true
+  })
+  it('should fail to validate on non equal values', () => {
+    const stubRemoteSecret: DroneSecret = {
+      clientId: Buffer.from('abc').toString('base64'),
+      clientSecret: Buffer.from('def').toString('base64'),
+    }
+    const stubOauthApps: OAuth2Application[] = [
+      {
+        name: 'otomi-drone',
+        clientId: 'abcd',
+      },
+    ]
+    expect(isSecretValid(stubRemoteSecret, stubOauthApps)).to.be.false
+    expect(consoleLog.calledWith('OAuth data did not match with expected secret')).to.be.true
+  })
+})
+
+describe('Gitea-Drone: Parse CSRF Token', () => {
+  const cookieName = '_csrf'
+  it('should successfully validate the token', () => {
+    const stubHeaderCookies: string[] = [`${cookieName}:abc`]
+    expect(getCookieValue(stubHeaderCookies, cookieName)).to.be.equal('abc')
+  })
+  it('should successfully validate parse the token from string', () => {
+    const stubHeaderCookies: string[] = [`${cookieName}:abc;`]
+    expect(getCookieValue(stubHeaderCookies, cookieName)).to.be.equal('abc')
+  })
+  it('should successfully validate parse the token from string containing more key/values', () => {
+    const stubHeaderCookies: string[] = [`${cookieName}:abc;redkubes:otomi`]
+    expect(getCookieValue(stubHeaderCookies, cookieName)).to.be.equal('abc')
+  })
+  it('should fail without a CSRF cookie', () => {
+    const stubHeaderCookies: string[] = ['redkubes:otomi']
+    expect(() => getCookieValue(stubHeaderCookies, cookieName)).to.throw(
+      GiteaDroneError,
+      `No ${cookieName} cookie was returned`,
+    ) // Need to expect(()=>func) otherwise throw will fail
+  })
+  it('should fail without a CSRF token', () => {
+    const stubHeaderCookies: string[] = [`${cookieName}:;`]
+    expect(() => getCookieValue(stubHeaderCookies, cookieName)).to.throw(
+      GiteaDroneError,
+      `No value for ${cookieName} was found`,
+    ) // Need to expect(()=>func) otherwise throw will fail
+  })
+})
+
+describe('Gitea-Drone: Authorize OAuth', () => {
```

Comment:
sorry, but this should go also, as this is only testing axios

Reply:
There is no explicit logic in test.
But this test is checking if parsing is performed correctly.